src/modules/blockchainApplication/manage/store/reducer.js

```diff
@@ -36,20 +36,32 @@ export const pins = (state = initialState.pins, { type, chainId }) => {
 /**
  *
  * @param {Object} state
- * @param {type: String, data: Object} action
+ * @param {type: String, app: Object, apps: Object, chainId: String} action
  */
-export const applications = (state = initialState.applications, { type, application, chainId }) => {
+export const applications = (state = initialState.applications, { type, app, apps, chainId }) => {
```

Comment:
It is not advisable [to mutate the redux state ](https://redux.js.org/faq/immutable-data#why-does-a-reducer-mutating-the-state-prevent-react-redux-from-re-rendering-a-wrapped-component) this way. This whole reducer smells a bit to me, dont we have any noticeable bugs on the components utilising this state? 🤔

Reply:
hmmm nope we don't so we have refactored reducer handlers and all suitable actions to adjust to this